src/relcast.erl

```diff
@@ -714,77 +710,85 @@ handle_message(Key, CF, FromActorID, Message, Transaction, State = #state{module
-                    NewState1 = maybe_serialize(Module, ModuleState, NewState),
-                    {stop, Timeout, maybe_dirty(Dirty, NewState1)}
+                    {stop, Timeout, maybe_dirty(Dirty, NewState)}
             end
     end.
 
 %% write all resulting messages and keys in an atomic transaction
-handle_actions([], _Transaction, State) ->
+handle_actions(_, [], _Transaction, State) ->
     {ok, State};
-handle_actions([new_epoch|Tail], Transaction, State) ->
+handle_actions(Seq, [new_epoch|Tail], Transaction, State) ->
     ok = rocksdb:transaction_commit(Transaction),
     {ok, NewCF} = rocksdb:create_column_family(State#state.db, make_column_family_name(State#state.epoch + 1),
                                                State#state.db_opts),
     ok = rocksdb:drop_column_family(State#state.active_cf),
     ok = rocksdb:destroy_column_family(State#state.active_cf),
+    %% filter old floating acks
+    Floats = maps:map(fun(_K, Acks) ->
+                              lists:filter(fun({_Seq, Epoch}) ->
+                                                   Epoch /= State#state.epoch
+                                           end, Acks)
+                      end, State#state.floated_acks),
     %% when we're done handling actions, we will write the module state (and all subsequent outbound
     %% messages from this point on) into the active CF, which is this new one now
     {ok, Transaction1} = transaction(State#state.db, State#state.write_opts),
-    handle_actions(Tail, Transaction1, State#state{out_key_count=0, active_cf=NewCF,
-                                                   transaction = Transaction1,
-                                                   transaction_dirty = false,
-                                                   epoch=State#state.epoch + 1, pending_acks=#{}});
-handle_actions([{multicast, Message}|Tail], Transaction, State =
+    handle_actions(Seq, Tail, Transaction1, State#state{out_key_count=0, active_cf=NewCF,
+                                                        transaction = Transaction1,
+                                                        transaction_dirty = false,
+                                                        floated_acks = Floats,
+                                                        epoch=State#state.epoch + 1, pending_acks=#{}});
+handle_actions(Seq, [{multicast, Message}|Tail], Transaction, State =
                #state{out_key_count=KeyCount, bitfieldsize=BitfieldSize, id=ID, ids=IDs, active_cf=CF, module=Module}) ->
     Bitfield = make_bitfield(BitfieldSize, IDs, ID),
     Key = make_outbound_key(KeyCount),
-    ok = rocksdb:transaction_put(Transaction, CF, Key, <<0:2/integer, Bitfield:BitfieldSize/bits, Message/binary>>),
+    ok = rocksdb:transaction_put(Transaction, CF, Key, <<0:2/integer, Bitfield:BitfieldSize/bits,
+                                                         Seq:32/integer-unsigned-big, Message/binary>>),
```

Comment:
I don't understand why we need to put the Seq in the database? Can't we just keep a list of outbound keys we've generated since the last sync, and check the `taken` keys against that?